test/extended/router/gatewayapi.go

```diff
@@ -0,0 +1,168 @@
+package router
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = g.Describe("[sig-network][OCPFeatureGate:GatewayAPI][Feature:Router][apigroup:gateway.networking.k8s.io]", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc       = exutil.NewCLIWithPodSecurityLevel("gateway-api", admissionapi.LevelBaseline)
+		crdNames = []string{
+			"gatewayclasses.gateway.networking.k8s.io",
+			"gateways.gateway.networking.k8s.io",
+			"httproutes.gateway.networking.k8s.io",
+			"referencegrants.gateway.networking.k8s.io",
+		}
+		errorMessage = "ValidatingAdmissionPolicy 'openshift-ingress-operator-gatewayapi-crd-admission' with binding 'openshift-ingress-operator-gatewayapi-crd-admission' denied request: Gateway API Custom Resource Definitions are managed by the Ingress Operator and may not be modified"
+	)
+
+	g.Describe("The Gateway API CRDs", func() {
+		g.It("should already be installed", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Get and check the installed CRDs")
+			for i := range crdNames {
+				crd, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), crdNames[i], metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				e2e.Logf("Found the Gateway API CRD named: %v", crd.Name)
+			}
+		})
+
+		g.It("should not be deleted", func() {
+			ctx := context.Background()
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Trying to delete the CRDs")
+			for i := range crdNames {
+				err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Delete(ctx, crdNames[i], metav1.DeleteOptions{})
+				o.Expect(err).To(o.HaveOccurred())
+				e2e.Logf("Got error: %v", err.Error())
+				o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
```

Comment:
I suppose that `Expect()` will log the error if it does not match the expected one. Therefore we can remove the log before, to be less verbose.

Reply:
Agree, I removed the log from the three cases. Thank you!